server/src/test/java/org/elasticsearch/cluster/metadata/IndexCreationTaskTests.java

```diff
@@ -429,7 +429,14 @@ private ClusterState executeTask() throws Exception {
         setupRequest();
         final MetaDataCreateIndexService.IndexCreationTask task = new MetaDataCreateIndexService.IndexCreationTask(
             logger, allocationService, request, listener, indicesService, aliasValidator, xContentRegistry, clusterStateSettings.build(),
-            validator, IndexScopedSettings.DEFAULT_SCOPED_SETTINGS);
+            validator, IndexScopedSettings.DEFAULT_SCOPED_SETTINGS) {
+
+            @Override
+            protected void checkShardLimit(final Settings settings, final ClusterState clusterState) {
```

Comment:
Could we add a short comment here explaining why this is here?

Reply:
I pushed f1ea945a372883f0e201746c0a5aa3312c01645c.